banjo/sema-expr.cpp

```diff
@@ -196,6 +196,12 @@ Parser::on_tuple_expression(Expr_list& es)
 Expr&
 Parser::on_dot_expression(Expr& e, Name& n)
 {
+  // Check to see if the type of e has been parsed yet.
+  // If it has not then we elaborate the type.
+  Type& t1 = e.type().non_reference_type();
+  if(is<Unparsed_type>(t1)){
```

Comment:
I am not sure if this is the correct place to elaborate the unparsed type, but without this qualified_lookup always threw an wrong type error since it was not a user-defined type. Note this does not solve the issue since make_member_reference now reaches a lingo_unreachable() statement